### PR TITLE
DDCNLS-7080 Redirect URI added to GG Route

### PR DESCRIPTION
**Pull request by RobertLaverick**

### CODE REVIEW

File: app/views/main.scala.html [hunk 1]
```diff
@@ -23,6 +23,7 @@
 @import controllers.auth.requests.UserRequest
 @import controllers.helpers.WelshWarningHelper
 @import util.LocalPartialRetriever
+@import java.net.URLEncoder
```
Comment (oc-ste) [inline app/views/main.scala.html]: Don't need this

File: app/views/main.scala.html [hunk 2]
```diff
@@ -136,7 +137,15 @@
...
-                profileUrl = request.profile,
+                profileUrl  = request.profile match {
+                    case Some(a) => {
+                        if (!a.contains("redirect_uri"))
+                            Some(a + "?redirect_uri=" + URLEncoder.encode(routes.HomeController.index().absoluteURL()))
+                        else
+                            Some(a)
+                        }
+                    case None => None
+                },
```
Comment (oc-ste) [inline app/views/main.scala.html]: profileUrl = request.profile.map(url => if(url.contains("redirect_uri")) {
url + "?redirect_uri=" + configDecorator.get(pertax-frontend) + (routes.HomeController.index())
else {
url 
}
}

File: test/controllers/auth/AuthActionSpec.scala [hunk 3]
```diff
@@ -266,4 +266,44 @@ class AuthActionSpec extends FreeSpec with MustMatchers with MockitoSugar with O
         s"Some(TrustedHelper(principalName,attorneyName,returnUrl,$fakePrincipalNino))")
     }
   }
+
+  "A user with a SCP Profile Url must include a redirect uri back to the home controller" in {
```
Comment (kurtlogan) [inline test/controllers/auth/AuthActionSpec.scala]: What would happen if you provided an invalid url?

File: app/views/main.scala.html [hunk 4]
```diff
@@ -136,7 +136,7 @@
                 )),
                 signoutUrl = Some(controllers.routes.ApplicationController.signout(Some(SafeRedirectUrl(configDecorator.getFeedbackSurveyUrl(configDecorator.defaultOrigin))), None).url),
                 activeTab = request.activeTab,
-                profileUrl = request.profile,
+                profileUrl  = request.profile,
```
Comment (kurtlogan) [inline app/views/main.scala.html]: Undo this change

File: test/controllers/auth/AuthActionSpec.scala (hunk 3 above)
Comment (RobertLaverick) [reply test/controllers/auth/AuthActionSpec.scala]: I've put together a fix for it the URL has a scheme as a way to check for an absolute URL, but I'm 100% sure that the SCP Profile Urls in prod are absolute or relative.

Comment (RobertLaverick) [reply test/controllers/auth/AuthActionSpec.scala]: I've had this confirmed by #team-scp-si. It's always absolute

Comment (kurtlogan) [reply test/controllers/auth/AuthActionSpec.scala]: We should assume they can be both

Comment (kurtlogan) [reply test/controllers/auth/AuthActionSpec.scala]: Thats fine the way it is then
